tests/test_wallet.py

```diff
@@ -1195,6 +1195,43 @@ def test_hsmtool_dump_descriptors(node_factory, bitcoind):
     assert len(bitcoind.rpc.listunspent(1, 1, [addr])) == 1
 
 
+@pytest.mark.openchannel('v1')
+def test_hsmtool_guess_to_remote_non_anchor(node_factory, bitcoind):
+    l1, l2 = node_factory.get_nodes(2)
+    # Allow l2 some warnings
+    l2.allow_warning = True
+    amount = 500000
+    l1.fundwallet(20000000)
+
+    l1.rpc.connect(l2.info['id'], 'localhost', l2.port)
+    l1.rpc.fundchannel(l2.info['id'], amount)
+    bitcoind.generate_block(6)
+    l1.daemon.wait_for_log('to CHANNELD_NORMAL')
+    wait_for(lambda: [c['active'] for c in l1.rpc.listchannels(l1.get_channel_scid(l2))['channels']] == [True, True])
+
+    # unilateral close channels l1<->l2
+    l1.stop()
+    l2.rpc.close(l1.info['id'], 1)
+
+    # Wait til to_self_delay expires, l1 should claim to_local back
+    bitcoind.generate_block(10, wait_for_mempool=1)
+    sync_blockheight(bitcoind, [l2])
+    log = l2.daemon.wait_for_log(r'Script to-them: ')
+    m = re.search(r'Script to-them: ([a-f0-9]{44})', log)
+    onchain_addr = bitcoind.rpc.decodescript(m.group(1))['address']
```

Comment:
Test is failing with a keyError on line 1221. Dict returned looks like:

`{'asm': '0 [error]', 'type': 'nonstandard', 'p2sh': '2MuBEKJ9vRmy64SSSRheWzf7np7Tcjj1Zgw', 'segwit': {'asm': '0 29e481065c0f83c94cec07ee31c833471c151989d405e77a0c39812048931bb5', 'hex': '002029e481065c0f83c94cec07ee31c833471c151989d405e77a0c39812048931bb5', 'address': 'bcrt1q98jgzpjup7pujn8vqlhrrjpnguwp2xvf6sz7w7sv8xqjqjynrw6sxvm489', 'type': 'witness_v0_scripthash', 'p2sh-segwit': '2NBGWnH9LN5ZqdQ5sU1QPdkjfEhM1Wfb7kS'}}`

should this be `bitcoind.rpc.decodescript(m.group(1))['segwit']['address']`?

Reply:
hsmtool invokes okay with this change:
`DEBUG:root:Starting 'tools/hsmtool guesstoremote bcrt1q98jgzpjup7pujn8vqlhrrjpnguwp2xvf6sz7w7sv8xqjqjynrw6sxvm489 022d223620a359a47ff7f7ac447c85c46c923da53389221a0054c11c1e3ca31d59 100 /tmp/ltests-_vq9dgd1/test_hsmtool_guess_to_remote_non_anchor_1/lightning-1/regtest/hsm_secret'`
But returns
`Could not find any basepoint matching the provided witness programm.
Are you sure that the channel used option_static_remotekey ?`